examples/vision/xray_classification_with_tpus.py

```diff
@@ -0,0 +1,479 @@
+    x = tf.keras.layers.Conv2D(16, 3, activation="relu", padding="same")(x)
+    x = tf.keras.layers.Conv2D(16, 3, activation="relu", padding="same")(x)
+    x = tf.keras.layers.MaxPool2D()(x)
+
+    x = conv_block(32)(x)
+    x = conv_block(64)(x)
+
+    x = conv_block(128)(x)
+    x = tf.keras.layers.Dropout(0.2)(x)
+
+    x = conv_block(256)(x)
+    x = tf.keras.layers.Dropout(0.2)(x)
+
+    x = tf.keras.layers.Flatten()(x)
+    x = dense_block(512, 0.7)(x)
+    x = dense_block(128, 0.5)(x)
+    x = dense_block(64, 0.3)(x)
+
+    outputs = tf.keras.layers.Dense(1, activation="sigmoid")(x)
+
+    model = tf.keras.Model(inputs=inputs, outputs=outputs)
+
+    return model
+
+
+"""
+## Correct for data imbalance
+
+We saw earlier in this notebook that the data was imbalanced, with more images classified
+as pneumonia than normal. We will correct for that in this following section.
+"""
+
+initial_bias = np.log([COUNT_PNEUMONIA / COUNT_NORMAL])
+print("Initial bias: {:.5f}".format(initial_bias[0]))
+
+weight_for_0 = (1 / COUNT_NORMAL) * (TRAIN_IMG_COUNT) / 2.0
+weight_for_1 = (1 / COUNT_PNEUMONIA) * (TRAIN_IMG_COUNT) / 2.0
+
+class_weight = {0: weight_for_0, 1: weight_for_1}
+
+print("Weight for class 0: {:.2f}".format(weight_for_0))
+print("Weight for class 1: {:.2f}".format(weight_for_1))
+
+"""
+The weight for class `0` (Normal) is a lot higher than the weight for class `1`
+(Pneumonia). Because there are less normal images, each normal image will be weighted
+more to balance the data as the CNN works best when the training data is balanced.
+"""
+
+"""
+## Train the model
+
+Since there are only two possible labels for the image, we will be using the
+`binary_crossentropy` loss. When we fit the model, identify the class weights. Because we
+are using a TPU, training will be relatively quick.
+
+For our metrics, we want to include precision and recall as they will provide use with a
+more informed picture of how good our model is. Accuracy tells us what fractions are the
+labels are correct. Since our data is not balanced, accuracy might give a skewed sense of
+a good model (i.e. a model that always predicts PNEUMONIA will be 74% accurate but is not
+a good model).
+
+Precision is the number of true positives (TP) over the sum of TP and false positives
+(FP). It shows what fraction of labeled positives are actually correct.
+
+Recall is the number of TP over the sum of TP and false negatves (FN). It shows what
+fraction of actual positives are correct.
+"""
+
+with strategy.scope():
+    model = build_model()
+
+    METRICS = [
+        "accuracy",
+        tf.keras.metrics.Precision(name="precision"),
+        tf.keras.metrics.Recall(name="recall"),
+    ]
+
+    model.compile(optimizer="adam", loss="binary_crossentropy", metrics=METRICS)
+
+"""
+From exploring the data and the model, I noticed that the training for the model has a
+slow start. However, after 25 epochs, the model slowly starts to converge.
+"""
+
+"""
+## Finetune the model
```

Comment:
You probably want to adjust the copy of this section to match the code. Also, keep in mind that fine-tuning is something that you do on a different dataset (transfer learning), so it's not applicable in this context